OptimizelySDKCore/OptimizelySDKCore/OPTLYEventBuilder.m

```diff
@@ -14,114 +14,113 @@
  * limitations under the License.                                           *
  ***************************************************************************/
 
-#import "OPTLYAttribute.h"
-#import "OPTLYBucketer.h"
-#import "OPTLYDecisionEventTicket.h"
-#import "OPTLYDecisionService.h"
-#import "OPTLYErrorHandler.h"
+#import "OPTLYControlAttributes.h"
 #import "OPTLYEvent.h"
 #import "OPTLYEventBuilder.h"
 #import "OPTLYEventFeature.h"
-#import "OPTLYEventDecision.h"
-#import "OPTLYEventDecisionTicket.h"
-#import "OPTLYEventHeader.h"
-#import "OPTLYEventLayerState.h"
 #import "OPTLYEventMetric.h"
 #import "OPTLYEventParameterKeys.h"
 #import "OPTLYExperiment.h"
 #import "OPTLYLogger.h"
 #import "OPTLYProjectConfig.h"
 #import "OPTLYVariation.h"
-#import "OPTLYControlAttributes.h"
 
 NSString * const OptimizelyActivateEventKey = @"campaign_activated";
 
 // --- Event URLs ----
 NSString * const OPTLYEventBuilderEventsTicketURL   = @"https://logx.optimizely.com/v1/events";
 
-@implementation OPTLYEventBuilderDefault : NSObject 
+@interface OPTLYEventBuilderDefault ()
+
+@property (readonly, strong) OPTLYProjectConfig *config;
+
+@end
+
+@implementation OPTLYEventBuilderDefault : NSObject
+
+-(instancetype)initWithConfig:(OPTLYProjectConfig *)config {
+    self = [super init];
+    if (self != nil) {
+        _config = config;
+    }
+    return self;
+}
 
 // NOTE: A dictionary is used to build the decision event ticket object instead of
 // OPTLYDecisionEventTicket object to simplify the logic. The OPTLYEventFeature value can be a
 // string, double, float, int, or boolean.
 // The OPTLYJSONModel cannot support a generic primitive/object type, so each event tag
 // value would have to be manually checked and converted to the appropriate OPTLYEventFeature type.
--(NSDictionary *)buildImpressionEventTicket:(OPTLYProjectConfig *)config
-                                     userId:(NSString *)userId
-                              experimentKey:(NSString *)experimentKey
-                                variationId:(NSString *)variationId
-                                 attributes:(NSDictionary<NSString *,NSString *> *)attributes {
-    if (!config) {
-        return nil;
-    }
-    
-    if ([userId length] == 0) {
-        [config.logger logMessage:OPTLYLoggerMessagesUserIdInvalid withLevel:OptimizelyLogLevelWarning];
+-(NSDictionary *)buildImpressionEventTicketForUser:(NSString *)userId
+                                        experiment:(OPTLYExperiment *)experiment
+                                         variation:(OPTLYVariation *)variation
+                                        attributes:(NSDictionary<NSString *,NSString *> *)attributes {
+    if (!self.config) {
         return nil;
     }
     
-    if ([experimentKey length] == 0) {
-        [config.logger logMessage:OPTLYLoggerMessagesExperimentKeyInvalid withLevel:OptimizelyLogLevelWarning];
+    if ([OPTLYEventBuilderDefault isEmptyString:userId]) {
```

Comment:
I think these conditions must be placed before calling EventBuilder methods. it's an internal class that should be used only to generate payloads, not to check or validate conditions except attributes.